src/eventsource.js

```diff
@@ -929,10 +935,11 @@
             abortController = undefined;
           }
         } else {
+          var nextHeartbeat = Math.max((wasActivity || Date.now()) + heartbeatTimeout - Date.now(), 1);
```

Comment:
let me explain the components of my original formula:
1. wasActivity || Date.now()   -- the left part is for the case where wasActivity is null and abortController is undefined. This leads to a full interval of length heartbeatTimeout. If there was an activity, this gives us the time of the last activity. I call the result _lastActivity_ in the next step.
2. lastActivity + heartbeatTimeout - Date.now()   -- this gives us the _future_ part of the heartbeatTimeout interval, the part that has already _passed_ is (now - lastActivity). 
3. Math.max(futurePart, 1)   -- makes sure that there is never an endless loop by having a timeout of 0 or less for whatever reason (e.g. timer granularity, ...) there may be.

The formula that you suggested is essentially the same because Math.max(Date.now(), wasActivity) can (actually _must_ because of (1)) be reduced to Date.now(). Then it is only written as  _b - (c - a)_  versus   _a + b - c_

Reply:
@jtempl, thank you for the explanation,
I am giving the same formula just to have the  `c` and `a` closer. And to show that Date.now() is only used to compute the elapsed time (and so the wall clock correctness is not used). `Date.now() - wasActivity` may be negative in case of "clock jumps", although, this is a rare case.